ministryofjustice/cloud-platform-cli
# feat ✨: Add a cluster command and a node-recycler sub-command

Overview
---

This PR connects to ministryofjustice/cloud-platform#3361 and relates to the requirement for a node recycler to run in an EKS cluster. We already have another Ruby script that does the same thing for kOps specific clusters. I wanted to move away from this Ruby script for a number of reasons, but mainly because the Cloud Platform team have adopted Go. Adding the node-recycle command to the cloud-platform cli also enables everyone with access to an AWS profile the ability to run a single command to recycle.

What's in this PR
---

This PR contains a new package `recycle` which contains the code required to run the below command. It utilises the `client` and `cluster` packages that have recently merged.

```
cloud-platform cluster recycle-node

Usage:
  cloud-platform cluster recycle-node [flags]

Examples:
$ cloud-platform cluster recycle-node


Flags:
      --aws-profile string   aws profile to use (default "default")
      --aws-region string    aws region to use (default "eu-west-2")
      --debug                enable debug logging
  -h, --help                 help for recycle-node
      --kubecfg string       path to kubeconfig file (default "/Users/jasonbirchall/.kube/config")
  -n, --name string          name of node to recycle
      --oldest               whether to recycle the oldest node
  -t, --timeout int          amount of time to wait for the drain command to complete (default 360)
```

How to test this
---

Checkout the branch and run:

```
go test -v ./pkg/recycle/...
```

or

```
go run ./cmd/main.go cluster recycle-node --skip-version-check --oldest
```
note: This will recycle the oldest node on the cluster in your current context.
